pkg/credentialprovider/provider.go

```diff
@@ -33,15 +33,16 @@ type DockerConfigProvider interface {
 	Enabled() bool
 	// Provide returns docker configuration.
 	// Implementations can be blocking - e.g. metadata server unavailable.
```

Comment:
I think we should update the comments here for `Provide`. Maybe something like:
```
// Provide returns docker configuration.
// Implementations can be blocking - e.g. metadata server unavailable.
// The image is passed into Provide in the event that the provider depends on any information
// in the image URL (e.g. the region where the repo is). The image can be ignored otherwise.
```

Reply:
In general I agree but think that the parantetical is a little to leading to the AWS use-case. Would instead propose:

```
// Provide returns docker configuration.
// Implementations can be blocking - e.g. metadata server unavailable.
// The image is passed into Provide as context in the event that the implementation depends 
// on information in the URL to return credentials, implementations are safe to ignore this.
```

Reply:
sgtm!